### PR TITLE
Perception: Add cascaded camera tracker

### PR DESCRIPTION
**Pull request by eugenegx**

### CODE REVIEW

File: modules/perception/obstacle/camera/tracker/cascaded_camera_tracker.cc [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    dlf_tracker_.GetAffinityMatrix(img, tracks_, detected,
+                                   &dlf_affinity_matrix);
+
+    // Merge
+    MergeAffinityMatrix(dlf_affinity_matrix, &affinity_matrix);
+
+    // High confidence selection filtering. Thresholds are tuned in 3 passes
+    // For entry higher than threshold, see if there is other competing
+    // high scores in the cross (same row or column)
+    FilterAffinityMatrix(0.95f, 0.95f, &affinity_matrix);
```
Comment (Capri2014) [inline modules/perception/obstacle/camera/tracker/cascaded_camera_tracker.cc]: may want to change this one to proto configs later for easier tuning.

File: modules/perception/obstacle/camera/tracker/cascaded_camera_tracker_util.cc [hunk 2]
```diff
@@ -79,8 +74,8 @@ void MergeAffinityMatrix(const std::vector<std::vector<float>> &to_merge,
 
   for (size_t i = 0; i < affinity_matrix->size(); ++i) {
     for (size_t j = 0; j < (*affinity_matrix)[0].size(); ++j) {
-      if ((*affinity_matrix)[i][j] <
-          9.0f) {  // For saving high confidence selection
+      // For saving high confidence selection
+      if ((*affinity_matrix)[i][j] < 9.0f) {
```
Comment (Capri2014) [inline modules/perception/obstacle/camera/tracker/cascaded_camera_tracker_util.cc]: Also this one

**Comment by eugenegx** [comment]
Thank you guys!

File: modules/perception/obstacle/camera/tracker/cascaded_camera_tracker.cc (hunk 1 above)
Comment (eugenegx) [reply modules/perception/obstacle/camera/tracker/cascaded_camera_tracker.cc]: Yes, will do it later when tuning is needed

**Pull request merged**